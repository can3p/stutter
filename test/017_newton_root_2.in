(def newton-root-2 (λ (x)
                      (/ (+ x (/ 2 x)) 2)))

(def newton-root-finder
     (λ (guess iterations)
        (if (= iterations 0)
            (list guess)
          (cons guess (newton-root-finder (newton-root-2 guess) (- iterations 1))))))

;;
;; Find the √2 using newton's method.
;;
(newton-root-finder 2.0 4)
