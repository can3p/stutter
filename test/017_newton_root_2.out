(2.0000000000 1.5000000000 1.4166666667 1.4142156863 1.4142135624)
